### PR TITLE
Added sort by year option

### LINKED ISSUE #22: [Feature Request] Sort library by year/date
Right now you can sort your library by author, title, and duration. Adding sort by year would be very valuable

### PR DESCRIPTION
**Pull request by lks-nbg**
This extension allows sorting the library by year
Resolves #22

### CODE REVIEW

File: app/src/main/java/io/github/mattpvaughn/chronicle/data/model/Audiobook.kt [hunk 1]
```diff
@@ -64,6 +65,7 @@ data class Audiobook constructor(
             parentId = dir.parentRatingKey,
             genre = dir.plexGenres.joinToString(separator = ", "),
             summary = dir.summary,
+            year = dir.parentYear.takeIf { it != 0} ?: dir.year,
```
Comment (mattttvaughn) [inline app/src/main/java/io/github/mattpvaughn/chronicle/data/model/Audiobook.kt]: What is the parentYear of a book? Depending on what that is, thinking we probably want to take dir.year as primary here and use parentYear as the ballback

**Review by mattttvaughn — CHANGES_REQUESTED** [review]
Looks good! Left one little comment

Also, for this we'd need a DB migration strategy for the change to Audiobook since that model is used in BookDatabase.

Should be quick n ez- you can just follow the existing patterns in `BookDatabase.kt` to add a new migration, and make sure to increment the version number on line 76

File: app/src/main/java/io/github/mattpvaughn/chronicle/data/model/Audiobook.kt (hunk 1 above)
Comment (lks-nbg) [reply app/src/main/java/io/github/mattpvaughn/chronicle/data/model/Audiobook.kt]: I found it in the list Metadata of MediaContainer, where the PlexDirectories are extracted. But I agree year should be used as primary and parentYear as fallback.

**Comment by lks-nbg** [comment]
- Added db migration code
- Changed year to primary and parentYear to fallback

**Review by mattttvaughn — APPROVED** [review]
looks good 👍

**Comment by pmayeur** [comment]
woot! Thank you! (If you provide an apk, I'm happy to QE)

**Pull request merged**